src/API/IJobs.cs

```diff
@@ -37,14 +38,20 @@ public interface IJobs
         /// <param name="pipelineId">Pipeline ID to create a new job from.</param>
         /// <param name="jobName">Name of the job.</param>
         /// <param name="jobPriority">Priority of the job.</param>
-        Task<Job> Create(string pipelineId, string jobName, JobPriority jobPriority);
+        Task<Job> Create(string pipelineId, string jobName, JobPriority jobPriority, IDictionary<string,string> metadata);
 
         /// <summary>
         /// Starts the job
         /// </summary>
         /// <param name="job">Job to start.</param>
         Task Start(Job job);
 
+        /// <summary>
+        /// Starts the job
+        /// </summary>
+        /// <param name="job">Add metadata to a job.</param>
+        Task AddMetadata(Job job, Dictionary<string, string> metadata);
```

Comment:
nit: interfaces are generally better served by the least common type such as `Dictionary<K,V>` -> `IReadOnlyDictionary<K,V>`

Reply:
Will fix in upcoming PR